code/online_challenges/src/codechef/COINS/COINS.py

```diff
@@ -0,0 +1,25 @@
+a = {};
+
+def dollar(n):
+    
+    if(n<=11):
+        a[n]=n;
+        return a[n];
+        
+    if n in a:
+        return a[n];
+        
+    a[n] = max(n, dollar(int(n/2))+dollar(int(n/3))+dollar(int(n/4)));
+    return a[n];
+
+while(True):
+    try:
+        n = int(input());
+    except EOFError:
```

Comment:
No need for the try catch.

Reply:
@arnavb, the input contains test cases that will continue till End of File, there is no predefined number of inputs. So to catch that error, we need to use try and catch. Please suggest me if anything else can be done to improve. Thank You!